### PR TITLE
Implement Brackets popup

### PR DESCRIPTION
**Pull request by anatoly-os**
TODO:
- Crash happens on my machine when incrementing value for Column or Span property

### CODE REVIEW

File: inspectors/models/abstractinspectormodel.cpp [hunk 1]
```diff
@@ -264,6 +269,10 @@ QVariant AbstractInspectorModel::valueFromElementUnits(const Ms::Pid& pid, const
         return strList.join(",");
     }
 
+    case Ms::P_TYPE::INT: {
```
Comment (anatoly-os) [inline inspectors/models/abstractinspectormodel.cpp]: Not sure whether it is needed. That was an attempt to fix the crash in slots processing.

File: inspectors/view/qml/notation/brackets/BracketSettings.qml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import "../../common"
+
+PopupViewButton {
+    id: root
+
+    property alias model: bracketPopup.model
+
+    icon: {
+        var icon = IconNameTypes.BRACKET
+        if (model && model.bracketType === BracketTypes.BRACKET_TYPE_BRACE)
```
Comment (anatoly-os) [inline inspectors/view/qml/notation/brackets/BracketSettings.qml]: It doesn't work, unfortunately :(

**Review by anatoly-os — COMMENTED** [review]
The problem is that two things don't work:

1. Crash happens when I change the values in the Bracket properties
Screenshot 2020-05-13 at 17 42 17
Screenshot 2020-05-13 at 17 42 15
Screenshot 2020-05-13 at 17 42 11

File: inspectors/view/qml/notation/brackets/BracketSettings.qml (hunk 2 above)
Comment (vpereverzev) [reply inspectors/view/qml/notation/brackets/BracketSettings.qml]: it should be `model.bracketType.value`. You are comparing the "PropertItem" object with int

File: inspectors/view/qml/notation/brackets/BracketSettings.qml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            icon = IconNameTypes.BRACES
+
+        return icon
+    }
+
+    text: {
+        var text = qsTr("Bracket")
+        if (model && (model.bracketType === BracketTypes.BRACKET_TYPE_BRACE))
```
Comment (vpereverzev) [inline inspectors/view/qml/notation/brackets/BracketSettings.qml]: the same thing here, use .value of "PropertyItem" object

File: inspectors/models/abstractinspectormodel.cpp (hunk 1 above)
Comment (vpereverzev) [reply inspectors/models/abstractinspectormodel.cpp]: yes, this is not necessary, for simple cases like QVariant <int> or QVariant <QString> enough of what is in default

**Comment by anatoly-os** [comment]
@vpereverzev when I change selection while the Inspector is opened, `isBracketTypeEqual()` method is not called until I drop the selection. It leads to not changing the popup icon and text.

I tend to reimplement the brackets and braces as separate components to avoid all these "bugs".

**Comment by anatoly-os** [comment]
Ok, Bracket and Brace are now separate components and popups.

The crash still happens :(

File: inspectors/view/qml/notation/brackets/internal/BracketPopupSection.qml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import QtQuick 2.9
+import MuseScore.Inspectors 3.3
+
+import "../../../common"
+
+Column {
+
+    property QtObject intBracketProperty: undefined
+    property string labelProperty: undefined
```
Comment (vpereverzev) [inline inspectors/view/qml/notation/brackets/internal/BracketPopupSection.qml]: it's better to initialize string by empty string 
But actually you can just use an alias property
`property alias title: textLabel.text`

**Review by vpereverzev — CHANGES_REQUESTED** [review]

**Review by vpereverzev — APPROVED** [review]

**Comment by anatoly-os** [comment]
The crash happen if and only if one bracket is selected when we change the column or span property. If there is multiple selection of several brackets or a bracket with different elements, the crash won't happen.

**Pull request merged**